### PR TITLE
Prevent hardLinkFile from unlinking a target just to then turn around…

### PR DESCRIPTION
**Pull request by daym**
… and link the same source to the same target again (on Posix).

### CODE REVIEW

File: source/dub/internal/vibecompat/core/file.d [hunk 1]
```diff
@@ -145,8 +145,14 @@ version (Windows) extern(Windows) int CreateHardLinkW(in wchar* to, in wchar* fr
 // guess whether 2 files are identical, ignores filename and content
 private bool sameFile(Path a, Path b)
 {
-	static assert(__traits(allMembers, FileInfo)[0] == "name");
-	return getFileInfo(a).tupleof[1 .. $] == getFileInfo(b).tupleof[1 .. $];
+	version (Posix) {
+		auto st_a = std.file.DirEntry(a.toNativeString).statBuf;
+		auto st_b = std.file.DirEntry(a.toNativeString).statBuf;
+		return st_a == st_b;
```
Comment (MartinNowak) [inline source/dub/internal/vibecompat/core/file.d]: This does an identity check, but an identical copy would be OK as well.
How about just using the same getFileInfo heuristic?

File: source/dub/internal/vibecompat/core/file.d [hunk 2]
```diff
@@ -145,8 +145,14 @@ version (Windows) extern(Windows) int CreateHardLinkW(in wchar* to, in wchar* fr
 // guess whether 2 files are identical, ignores filename and content
 private bool sameFile(Path a, Path b)
 {
-	static assert(__traits(allMembers, FileInfo)[0] == "name");
-	return getFileInfo(a).tupleof[1 .. $] == getFileInfo(b).tupleof[1 .. $];
+	version (Posix) {
+		auto st_a = std.file.DirEntry(a.toNativeString).statBuf;
+		auto st_b = std.file.DirEntry(a.toNativeString).statBuf;
```
Comment (MartinNowak) [inline source/dub/internal/vibecompat/core/file.d]: Typo s/a/b/

**Comment by MartinNowak** [comment]
Why did you close it?

**Comment by daym** [comment]
Replaced by #1071 where I used a non-master branch so I can make more changes.

**Pull request closed without merging**